Rich_Text_Example.c

```diff
@@ -0,0 +1,270 @@
+
+   _apply_attribute((Eo*)data);
+}
+
+static void
+_gui_red_cb(void *data, const Efl_Event *event EINA_UNUSED)
+{
+   if (efl_ui_selectable_selected_get((Eo*)data))
+      efl2_text_style_foreground_color_set(attribute_factory, 255, 0, 0, 255);
+   else
+      efl2_text_style_foreground_color_set(attribute_factory, 0, 0, 0, 255);
+
+   _apply_attribute((Eo*)data);
+}
+
+static void
+_gui_30_cb(void *data, const Efl_Event *event EINA_UNUSED)
+{
+   if (efl_ui_selectable_selected_get((Eo*)data))
+      efl2_text_font_properties_font_size_set(attribute_factory, 30);
+   else
+      efl2_text_font_properties_font_size_set(attribute_factory, 15);
+
+   _apply_attribute((Eo*)data);
+}
+
+/****************************************************/
+
+
+
+
+
+/************ Application Logic ******************/
+
+static void
+_gui_quit_cb(void *data EINA_UNUSED, const Efl_Event *event EINA_UNUSED)
+{
+   efl_exit(0);
+}
+ 
+EAPI_MAIN void
+efl_main(void *data EINA_UNUSED, const Efl_Event *ev EINA_UNUSED)
+{
+   Eo *win, *box, *boxHor, *ui_text, *content;
+ 
+   win = efl_add(EFL_UI_WIN_CLASS, efl_main_loop_get(),
+                 efl_ui_win_type_set(efl_added, EFL_UI_WIN_TYPE_BASIC),
+                 efl_text_set(efl_added, "Hello World"),
+                 efl_ui_win_autodel_set(efl_added, EINA_TRUE));
+ 
+   efl_event_callback_add(win, EFL_UI_WIN_EVENT_DELETE_REQUEST, _gui_quit_cb, NULL);
+ 
+   box = efl_add(EFL_UI_BOX_CLASS, win,
+                efl_content_set(win, efl_added),
+                efl_gfx_hint_size_min_set(efl_added, EINA_SIZE2D(360, 240)));
+ 
+   ui_text = efl_add(EFL_UI_TEXT_CLASS, box,
+           efl_text_markup_set(efl_added, "Hello World"),
+           efl_text_multiline_set(efl_added, EINA_FALSE),
+           efl_gfx_hint_weight_set(efl_added, 1.0, 0.9),
+           efl_gfx_hint_align_set(efl_added, 0.5, 0.5),
+           efl_event_callback_add(efl_added, EFL2_TEXT_RAW_EDITABLE_EVENT_SELECTION_CHANGED, // correct name ??
+                                  _gui_selection_changed_cb, efl_added),
+           efl_event_callback_add(efl_added, EFL_TEXT_CURSOR_EVENT_CURSOR_CHANGED,   // this event attached to the cursor, will it be received by the user ??
+                                  _gui_cursor_changed_cb, efl_added),
+           efl_pack(box, efl_added));
+
+   boxHor = efl_add(EFL_UI_BOX_CLASS, win,
+                efl_gfx_hint_weight_set(efl_added, 1.0, 0.1),
+                efl_ui_layout_orientation_set(efl_added, EFL_UI_LAYOUT_ORIENTATION_HORIZONTAL),
+                efl_pack(box, efl_added));
+
+   attribute_factory = efl_add(EFL2_TEXT_ATTRIBUTE_FACTORY, efl_main_loop_get());
+   cursor_main = efl2_text_raw_editable_main_cursor_get(ui_text);
+   cursor_start = efl_add(EFL2_TEXT_CURSOR, ui_text);
+   cursor_end = efl_add(EFL2_TEXT_CURSOR, ui_text);
+   efl2_text_raw_editable_selection_cursors_get(ui_text, &sel_start, &sel_end); // Can be called on ui_text ?

```

Comment:
getting sel_start,sel_end cursor in advance, where these cursors objects will not change. this is valid ?